include/circt/Dialect/RTL/RTLStructure.td

```diff
@@ -19,7 +19,8 @@ def RTLModuleOp : RTLOp<"module",
     name, a list of ports, and a body that represents the connections within
     the module.
   }];
-  let arguments = (ins StrArrayAttr:$argNames, StrArrayAttr:$resultNames);
+  let arguments = (ins StrArrayAttr:$argNames, StrArrayAttr:$resultNames, 
+                       OptionalAttr<StrAttr>:$verilogName);
```

Comment:
The `verilogName` extension is logically independent of bind support and can be broken out to its own patch.

That said, why do we need this?  The reason we have `verilogName` on ext module and generated module is because we don't support parameterized modules correctly.  When we do, we will eliminate this attribute and just use the module symbol.

Given that we can't *declare* a parameterized module at all right now, I don't think RTLModuleOp needs to carry this.

Reply:
The addition of the `getVerilogModuleNameAttr` is a useful cleanup though!